ui/packages/tidb-dashboard-lib/src/apps/Deadlock/pages/List.tsx

```diff
@@ -0,0 +1,110 @@
+import { DeadlockModel } from '@lib/client'
+import {
+  AnimatedSkeleton,
+  AutoRefreshButton,
+  Card,
+  CardTable
+} from '@lib/components'
+import openLink from '@lib/utils/openLink'
+import { useMemoizedFn } from 'ahooks'
+import { CacheContext } from '@lib/utils/useCache'
+import React, { useMemo, useState, useContext } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { useEffectOnce } from 'react-use'
+import { useTranslation } from 'react-i18next'
+import { DeadlockContext } from '../context'
+
+function List() {
+  const ctx = useContext(DeadlockContext)
+
+  const { t } = useTranslation()
+  const cache = useContext(CacheContext)
+  let [isLoading, setIsLoading] = useState(true)
+  let [items, setItems] = useState([] as DeadlockModel[])
+  const navigate = useNavigate()
+
+  const pullItems = async () => {
+    cache?.clear()
+    setIsLoading(true)
+    const { data } = await ctx!.ds.deadlockListGet()
+    data.forEach((it) => {
+      let items = cache?.get(`deadlock-${it.id}`) || []
+      items.push(it)
+      cache?.set(`deadlock-${it.id}`, items)
+    })
+    setItems(data)
+    setIsLoading(false)
+  }
+  const handleRowClick = useMemoizedFn(
+    (record, index, ev: React.MouseEvent<HTMLElement>) => {
+      openLink(`/deadlock/detail?id=${record.id}`, ev, navigate)
+    }
+  )
+  useEffectOnce(() => {
+    setIsLoading(true)
+    cache?.clear()
+    ctx!.ds
+      .deadlockListGet()
+      .then((res) => {
+        setItems(res.data)
+        res.data.forEach((it) => {
+          let items = cache?.get(`deadlock-${it.id}`) || []
+          items.push(it)
+          cache?.set(`deadlock-${it.id}`, items)
+        })
+      })
+      .catch((e) => {
+        console.error(e)
+      })
+      .finally(() => {
+        setIsLoading(false)
+      })
+  })
+  const summary = useMemo(() => {
+    let result = new Map()
+    for (const item of items) {
+      let summaryEntry = result.get(item.id) || {
+        id: item.id,
+        occur_time: item.occur_time,
+        items: []
+      }
+      summaryEntry.items.push(item)
+      result.set(item.id, summaryEntry)
+    }
+    return result
```

Comment:
Or return `Array.from(result.values())` directly, so you don't need to do it when rendering.